dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/event/DataValuesValidator.java

```diff
@@ -80,10 +81,20 @@ public void validate(Reporter reporter, TrackerBundle bundle, Event event) {
       validateDataValue(reporter, bundle, dataElement, dataValue, programStage, event);
     }
 
+    if (EventStatus.STATUSES_WITHOUT_DATA_VALUES.contains(event.getStatus())) {
```

Comment:
I would say keep the style of the methods below and move the if inside like in https://github.com/dhis2/dhis2-core/blob/f9d763cf8f351d3c22b9eab11f815fa6847cfc76/dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/event/DataValuesValidator.java#L150

so it reads

```java
validateDataValuesSetInAllowedEventStatus(reporter, event);
validateMandatoryDataValues(reporter, bundle, event);    validateDataValueDataElementIsConnectedToProgramStage(reporter, bundle, event, programStage);
```

We could also shorten the method names as we are in the `DataValuesValidator`. So for the new one maybe `validateEventStatus`.